engines/ultima/ultima8/kernel/kernel.cpp

```diff
@@ -324,7 +324,20 @@ void Kernel::save(Common::WriteStream *ws) {
 	_pIDs->save(ws);
 	ws->writeUint32LE(_processes.size());
 	for (ProcessIterator it = _processes.begin(); it != _processes.end(); ++it) {
-		(*it)->save(ws);
+		const char *cname = (*it)->GetClassType()._className; // virtual
+		uint16 clen = strlen(cname);
+
+		Std::map<Common::String, ProcessLoadFunc>::iterator iter;
+		iter = _processLoaders.find(cname);
+
+		if (iter == _processLoaders.end()) {
+			perr << "Process class cannot save without registered loader: " << cname << Std::endl;
+			continue;
```

Comment:
This would prevent a completely broken save but would be incomplete. Is it better to fail the save entirely or save the object even though load will fail?

Reply:
Skipping the process /object save causes other assertions to fail, so what's currently here is definitely not a good answer.